internal/display/bufwindow.go

```diff
@@ -526,6 +553,16 @@ func (w *BufWindow) displayBuffer() {
 						}
 					}
 
+					if b.Settings["altlinecolors"].(bool) {
+						//altlinecolors: apply alt line color on lines with selection
+						if (vloc.Y + w.StartLine.Line)%2 == 0 && selectionOnLine {
```

Comment:
This causes another bug: e.g. if `hlsearch` option is enabled and we've searched for some text and the search results are highlighted, then if we select some text in the same line that contains a search result, this search result becomes wrongly highlighted with `alt-line` instead of `hlsearch`.

This can be easily fixed by adding `&& !dontOverrideBackground` condition here. But as I noted in my above comment, I think we can and should remove this `selectionOnLine` logic altogether.